src/ScheduleFunctions.cpp

```diff
@@ -2262,25 +2282,38 @@ bool validate_schedule(Function f, const Stmt &s, const Target &target, bool is_
                 sites.erase(sites.begin() + possibly_invalid_idx);
                 // This one will also be invalid if we find a subsequent loop from the same func
                 possibly_invalid_idx = i;
-                store_idx = compute_idx = -1;
+                store_idx = compute_idx = hoist_storage_idx = -1;
             }
         }
     }
 
     // Check there isn't a parallel loop between the compute_at and the store_at
-    if (both_ok()) {
+    if (all_ok()) {
         for (int i = store_idx + 1; i <= compute_idx; i++) {
             if (sites[i].is_parallel) {
                 err << "Func \"" << f.name()
                     << "\" is stored outside the parallel loop over "
                     << sites[i].loop_level.to_string()
                     << " but computed within it. This is a potential race condition.\n";
-                store_idx = compute_idx = -1;
+                store_idx = compute_idx = hoist_storage_idx = -1;
+            }
+        }
+    }
+
+    // Check there isn't a parallel loop between the compute_at and the hoist_storage_at
+    if (all_ok()) {
+        for (int i = hoist_storage_idx + 1; i <= compute_idx; i++) {
+            if (sites[i].is_parallel) {
```

Comment:
is_parallel also covers vectorized and gpu loops, so the wording will need tweaking for the error message to not be confusing in those cases.